databases/experiment_database.py

```diff
@@ -1,445 +1,512 @@
+        '''Prints all experiment info to terminal.'''
+        self._c.execute("SELECT name, species, uses_rfid, num_animals, num_groups, cage_max FROM experiment")
+        print('Experiment: name, species, uses_rfid, num_animals, num_groups, cage_max')
+        print(self._c.fetchall())
+        self._c.execute("SELECT name, num_animals, full FROM groups")
+        print('Groups: name, num_animals, full')
+        print(self._c.fetchall())
+        self._c.execute("SELECT cage_id, group_id, num_animals, full FROM cages")
+        print('cages: cage_id, group_id, num_animals, full')
+        print(self._c.fetchall())
+        self._c.execute("SELECT item, auto FROM measurement_items")
+        print('Measurement Items: item, auto')
+        print(self._c.fetchall())
+
+    def close(self):
+        '''Closes database file.'''
+        self._conn.close()
+
+    def export_all_tables_to_csv(self, export_dir="exports"):
+        '''Exports all tables in the database to separate CSV files.'''
+        # Create directory to save CSV files if it doesn't exist
+        if not os.path.exists(export_dir):
+            os.makedirs(export_dir)
+
+        # Get the base name of the database file (without extension)
+        db_name = os.path.splitext(os.path.basename(self.db_file))[0]
+
+        # Create a directory with the database name to save CSV files
+        export_dir = os.path.join("exports", db_name)
+        if not os.path.exists(export_dir):
+            os.makedirs(export_dir)
+
+        # The variable `export_dir` now holds the path to the folder where the CSVs will be saved
+
+        # Get all table names from the database
+        self._c.execute("SELECT name FROM sqlite_master WHERE type='table';")
+        tables = self._c.fetchall()
+
+        # Export each table to a CSV file
+        for table_name in tables:
+            table_name = table_name[0]  # Extract table name from tuple
+            csv_file_path = os.path.join(export_dir, f"{table_name}.csv")
+
+            # Fetch all data from the table
+            self._c.execute(f"SELECT * FROM {table_name}")
+            rows = self._c.fetchall()
+
+            # Get column names for the table
+            self._c.execute(f"PRAGMA table_info({table_name})")
+            column_info = self._c.fetchall()
+            column_names = [col[1] for col in column_info]  # Column names are in the second position
+
+            # Write data to CSV
+            with open(csv_file_path, mode='w', newline='') as file:
+                csv_writer = csv.writer(file)
+                # Write the header
+                csv_writer.writerow(column_names)
+                # Write the rows
+                csv_writer.writerows(rows)
+
+            print(f"Table '{table_name}' exported to {csv_file_path}")
+
+        print("All tables have been exported successfully.")
+
+        # Once export is done, show the success notification
+        self.show_success_notification()

```

Comment:
Perhaps this function could return a success status, and the caller of export_all_tables_to_csv could show the success notification. This would allow you to remove the UI dependency from the database code.

Reply:
Here are two ideas:
As Dr. Holdener suggested, export_all_tables_to_csv should really just perform a single function -- exporting tables to csv. The UI logic should be handled in /experiment_pages/experiment/data_analysis_ui.py when export_to_csv is called. If you anticipate this UI element (the successful export popup) to be needed across multiple pages, then we should look into additional abstraction.